CONTRIBUTING.md

```diff
@@ -382,7 +438,6 @@ where the changes should be squashed:
 (Note that the message refers to the other commit's subject line, not the SHA-1. This way it's still
 relevant if there are intermediary rebases.)
 
-If you *really* need a newer commit from the base branch, or if the history is getting too
-complicated, it can be OK to force-push occasionally, provided that **all current reviewers agree**.
-
-Don't push a merge commit to a pull request under any circumstance.
+If you need new stuff from the base branch, it's fine to rebase and force-push, as long as you don't
+rewrite the history. Just give a heads up to the reviewers beforehand. Don't push a merge commit to
```

Comment:
If you use Github web UI to update the branch or resolve conflicts, it will create a merge commit. I personally don't think we absolutely must avoid merge commits in a PR.

Reply:
I find them really annoying, it's an extra set of changes that have nothing to do with the PR. Also they tend to get in the way when you do interactive rebases locally.